### PR TITLE
clisqlshell: improve the behavior of `\c`

### LINKED ISSUE #68302: cli: `\c` metacommand does not work in insecure mode
Filed upon request by @knz 

When running in insecure mode, switching the user with the `\c` CLI command doesn't work too well. E.g.

```
❯ ./cockroach sql --insecure
#
# Welcome to the CockroachDB SQL shell.
# All statements must be terminated by a semicolon.
# To exit, type: \q.
#
# Server version: CockroachDB CCL v21.2.0-alpha.00000000-2910-g701b177d8f (x86_64-apple-darwin20.6.0, built , go1.16.5) (same version as client)
# Cluster ID: ea4c23a1-5478-489f-a42b-880fc6e9b1cb
#
# Enter \? for a brief introduction.
#
root@:26257/defaultdb> \c - testuser
using new connection URL: postgresql://testuser@:26257/defaultdb
warning: error retrieving the transaction status: pq: SSL is not enabled on the server
warning: error retrieving the database name: pq: SSL is not enabled on the server
```

The problem seems to be that the `sslmode=disable` query param is not added

### PR DESCRIPTION
**Pull request by knz**
See individual commits for details.

Fixes #68302

### CODE REVIEW

**Review by rafiss — APPROVED** [review]
thank you for the improvements!

tcl tests are cool :)

File: pkg/server/pgurl/pgurl_test.go [hunk 1]
```diff
@@ -64,6 +64,49 @@ func TestURL(t *testing.T) {
...
+
+	enabled, hasp, p := u.GetAuthnPassword()
+	require.True(t, enabled)
+	require.True(t, hasp)
+	require.Equal(t, p, "abc")
+
+	u.ClearPassword()
+	enabled, hasp, p = u.GetAuthnPassword()
+	require.True(t, enabled)
+	require.False(t, hasp)
```
Comment (rafiss) [inline pkg/server/pgurl/pgurl_test.go:83]: does this need a `require.Equal(t, p, "")` ?

Comment (knz) [reply pkg/server/pgurl/pgurl_test.go:83]: if "haspassword" is false, there's no API guarantee about the remaining value in the password field itself.

**Comment by knz** [comment]
TFYR!

bors r=rafiss

**Pull request merged**